runutil/runutil_test.go

```diff
@@ -56,3 +63,107 @@ func (l *fakeLogger) Log(keyvals ...interface{}) error {
 	l.keyvals = keyvals
 	return nil
 }
+
+type testCloser = fakeCloser
+
+func TestCloseWithErrCapture(t *testing.T) {
+	for _, tcase := range []struct {
+		err    error
+		closer io.Closer
+
+		expectedErrStr string
+	}{
+		{
+			err:            nil,
+			closer:         testCloser{err: nil},
+			expectedErrStr: "",
+		},
+		{
+			err:            errors.New("test"),
+			closer:         testCloser{err: nil},
+			expectedErrStr: "test",
+		},
+		{
+			err:            nil,
+			closer:         testCloser{err: errors.New("test")},
+			expectedErrStr: "close: test",
+		},
+		{
+			err:            errors.New("test"),
+			closer:         testCloser{err: errors.New("test")},
+			expectedErrStr: "2 errors: test; close: test",
+		},
+	} {
+		if ok := t.Run("", func(t *testing.T) {
```

Comment:
This seems like a really odd way to structure this test (I assume this is a verbatim copy of the Thanos tests?)
* Conditionals in the test body for the various test cases instead of having three separate tests ala `t.Run("something", ...)`
* Using `t.Fail` and `t.Error` when we're already importing testify assert
* Using `t.Run()` with an empty test name?